src/map.cpp

```diff
@@ -914,18 +915,18 @@ uint16_t AStarNodes::getMapWalkCost(AStarNode* node, const Position& neighborPos
 	return MAP_NORMALWALKCOST;
 }
 
-uint16_t AStarNodes::getTileWalkCost(const Creature& creature, const Tile* tile)
+uint16_t AStarNodes::getTileWalkCost(std::shared_ptr<const Creature> creature, std::shared_ptr<const Tile> tile)
 {
 	uint16_t cost = 0;
-	if (tile->getTopVisibleCreature(&creature)) {
+	if (tile->getTopVisibleCreature(creature)) {
 		// destroy creature cost
 		cost += MAP_NORMALWALKCOST * 3;
 	}
 
-	if (const MagicField* field = tile->getFieldItem()) {
+	if (auto field = tile->getFieldItem()) {
 		CombatType_t combatType = field->getCombatType();
-		const Monster* monster = creature.getMonster();
-		if (!creature.isImmune(combatType) && !creature.hasCondition(Combat::DamageToConditionType(combatType)) &&
+		auto monster = creature->getMonster();
```

Comment:
Could this be `const auto`
```suggestion
		const auto monster = creature->getMonster();
```